### PR TITLE
cairo(-devel): fix crash on Big Sur

### PR DESCRIPTION
**Pull request by michaellass**
#### Description

Apply changes from https://gitlab.freedesktop.org/cairo/cairo/-/merge_requests/52 to fix crashes of several applications (gimp2, inkscape, meld, etc.) on macOS Big Sur.

See: https://gitlab.freedesktop.org/cairo/cairo/-/issues/420

Fixes: https://trac.macports.org/ticket/61586

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.1 20C69
Xcode 12.3 12C33

For me, this fixes crashes of gimp2 and meld that made those programs unusable before on Big Sur. According to the bug report, inkscape is affected as well by the same bug. However, I have not tested if the issue is resolved for inkscape.

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`? (cairo-test-suite hangs with 100% CPU for cairo. haven't tried for cairo-devel after that...)
- [ ] tried a full install with `sudo port -vst install`? (only without trace mode)
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by kencu** [comment]
so this has been a disaster since bigsur came out. hopefully this fixes it, finally

**Comment by michaellass** [comment]
I forgot to bump the revision, so I did that now. I also renamed the patch, trying to make `port lint --nitpick` happy.

**Comment by mascguy** [comment]
I was just bitten by this, when helping a friend setup his new Mac with MacPorts and GIMP.

@ryandesign, would you be interested in merging this?

**Comment by ryandesign** [comment]
Sorry, I did not see this PR until now. I am now reading about it.

**Comment by ryandesign** [comment]
> I also renamed the patch, trying to make `port lint --nitpick` happy.

This lint check is out of date and will be more relaxed in MacPorts 2.7.0. See https://trac.macports.org/ticket/59695

File: graphics/cairo-devel/Portfile [hunk 1]
```diff
@@ -101,6 +101,14 @@ platform macosx {
     # Don't allow Quartz support to be disabled. Keep the variant for awhile in
     # case any dependents are using the active_variants portgroup to check for it.
     variant_set             quartz
+
+    # Apply changes from https://gitlab.freedesktop.org/cairo/cairo/-/merge_requests/52
+    # on macOS Big Sur and newer. Fixes:
+    # https://gitlab.freedesktop.org/cairo/cairo/-/issues/420
+    # https://trac.macports.org/ticket/61586
+    if {${os.major} > 19} {
```
Comment (ryandesign) [inline graphics/cairo-devel/Portfile:109]: Why apply the patch conditionally? Upstream has merged the code unconditionally for all platforms; we should use it for all platforms too.

Comment (michaellass) [reply graphics/cairo-devel/Portfile:109]: The main reason is that I cannot test anything else than Big Sur and the old version didn't cause problems on older macOS versions. Feel free to apply it unconditionally.

**Pull request merged**